components/SearchDropdown/index.tsx

```diff
@@ -0,0 +1,86 @@
+import styles from "../../styles/SearchDropdown/dropdown.module.css";
+import { useState, useRef, useEffect } from "react";
+
+export default function SearchDropdown(props) {
```

Comment:
Can you create the type for the props here?

To do this you will also have to import the following

```
import { useState, useRef, useEffect, Dispatch, SetStateAction } from "react";
```

```suggestion


/**
 * The props for the SearchDropdown component
 */
interface Props {
    /**
     * The items to be show in the autocomplete
     */
    listItems: Array<string>,
    /**
     * The current value of the textbox
     */
    search: string,
    /**
     * The hook function to set the state for the component
     */
    setSearch: Dispatch<SetStateAction<string>>,
    /**
     * The keyValue
     */
    keyValue: string
}

export default function SearchDropdown(props: Props) {
```

Reply:
Okay, thanks